applications/nrf_desktop/src/modules/Kconfig.config

```diff
@@ -0,0 +1,28 @@
+#
+# Copyright (c) 2018 Nordic Semiconductor
+#
+# SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+#
+
+menu "Settings loading service"
+
+config DESKTOP_SETTINGS_LOAD_BY_THREAD_ENABLE
+	bool "Enable loading of setting by separate thead"
+	help
+	  The option enables loading of settings by separate
+	  thread in background instead of use of system
+	  work queue for that purpose.
+
+config DESKTOP_SETTINGS_LOAD_THREAD_STACK_SIZE
+	int "Settings module thread stack size"
+	range 0 4096
+	default 0
+	help
+	  Stack size for thread responsible for loading
+	  settings from flash memory.
+
+if DESKTOP_SETTINGS_LOAD_BY_THREAD_ENABLE
+DESKTOP_SETTINGS_LOAD_THREAD_STACK_SIZE=1024
```

Comment:
this is actually not the same as what I suggested in the comment
this will indeed set the value but will also make `DESKTOP_SETTINGS_LOAD_THREAD_STACK_SIZE`  appear in menuconfig even if `DESKTOP_SETTINGS_LOAD_BY_THREAD_ENABLE` is disabled.